### PR TITLE
Change terms Collator -> Notary to align with sharding 1.1 

### PR DESCRIPTION
**Pull request by terenc3t**
In sharding 1.1. Term **Collator** is depreciated, and replaced by **Notary**. To be consistent with the new terms, this PR addresses the collator to notary name changes across our sharding code base. As a purely cosmetic change, we are not fixing/introducing new features, the impact should be at the minimal

### CODE REVIEW

File: sharding/notary/notary_test.go [hunk 1]
```diff
@@ -71,60 +71,60 @@ func setup() (*backends.SimulatedBackend, *contracts.SMC) {
...
-	numCollators, err = smc.NumCollators(&bind.CallOpts{})
+	// Now there should be one notary
+	numNotaries, err = smc.NumNotaries(&bind.CallOpts{})
 	if err != nil {
 		t.Fatal(err)
 	}
-	if big.NewInt(1).Cmp(numCollators) != 0 {
-		t.Fatalf("Unexpected number of collators. Got %d, wanted 1.", numCollators)
+	if big.NewInt(1).Cmp(numNotaries) != 0 {
+		t.Fatalf("Unexpected number of notarys. Got %d, wanted 1.", numNotaries)
```
Comment (rauljordan) [inline sharding/notary/notary_test.go]: notaries*

File: sharding/notary/notary_test.go [hunk 2]
```diff
@@ -71,60 +71,60 @@ func setup() (*backends.SimulatedBackend, *contracts.SMC) {
...
-	numCollators, err := smc.NumCollators(&bind.CallOpts{})
+	// There should be no notarys initially
+	numNotaries, err := smc.NumNotaries(&bind.CallOpts{})
 	if err != nil {
 		t.Fatal(err)
 	}
-	if big.NewInt(0).Cmp(numCollators) != 0 {
-		t.Fatalf("Unexpected number of collators. Got %d, wanted 0.", numCollators)
+	if big.NewInt(0).Cmp(numNotaries) != 0 {
+		t.Fatalf("Unexpected number of notarys. Got %d, wanted 0.", numNotaries)
```
Comment (rauljordan) [inline sharding/notary/notary_test.go]: notaries*

**Comment by rauljordan** [comment]
Aside from the very minor typo, looks good

**Comment by terenc3t** [comment]
Thanks! I'll merge this after #93

**Pull request merged**